src/product/product.model.ts

```diff
@@ -0,0 +1,20 @@
+import { number, object, ObjectSchema, string } from 'yup'
+
+export interface Product {
+  id: string
+  title: string
+  description?: string
+  price: number
+  count: number
+}
+
+export interface ProductDto extends Omit<Product, 'id'> {}
```

Comment:
You could use:  
 `type ProductDto = InferType<typeof productSchema>;`
and 
`type Product = ProductDto & { id: string }` for example :)